### ISSUE TITLE
[0.60.2] import('cmake') cannot handle cmake not being on path. 

### ISSUE THREAD

**Issue by Neumann-A** [mesonbuild/meson#9733]
Meson just says it cannot find cmake although it has been passed via a native/cross file as: 
```
[binaries]
...
cmake = 'E:/vcpkg_cache/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/bin/cmake.exe'
....
```

CI run (https://dev.azure.com/vcpkg/public/_build/results?buildId=64620&view=logs&jobId=7922e5c4-0103-5f8f-ad17-45ce9bb98e80&j=878666d5-db33-5b27-9e7d-b0c7ee352005)

The fact that meson assumes stuff to be on path although full paths are passed seems to be a consistent error...... (same goes for ninja)

**Comment by mensinda** [comment]
I am unable to reproduce this. My setup: In `test cases/cmake/1 basic`; `meson build --native-file native.ini` with:

```ini
[binaries]

cmake = '/home/daniel/prefix/builds/cmake/bin/cmake'
```

I made sure to uninstall my system CMake and I have no other CMake installation in my `PATH`.

```
he Meson build system
Version: 0.60.2
Source dir: /home/daniel/projects/meson/test cases/cmake/1 basic
Build dir: /home/daniel/projects/meson/test cases/cmake/1 basic/build
Build type: native build
Project name: cmakeSubTest
Project version: undefined
C compiler for the host machine: cc (gcc 11.1.0 "cc (GCC) 11.1.0")
C linker for the host machine: cc ld.bfd 2.36.1
C++ compiler for the host machine: c++ (gcc 11.1.0 "c++ (GCC) 11.1.0")
C++ linker for the host machine: c++ ld.bfd 2.36.1
Host machine cpu family: x86_64
Host machine cpu: x86_64

Executing subproject cmMod method cmake

cmMod| Found CMake: /home/daniel/prefix/builds/cmake/bin/cmake (3.21.20210708)

| Configuring the build directory with CMake version 3.21.20210708
| Running CMake with: -G Ninja -DCMAKE_INSTALL_PREFIX=/usr/local
|   - build directory:          /home/daniel/projects/meson/test cases/cmake/1 basic/build/subprojects/cmMod/__CMake_build
|   - source directory:         /home/daniel/projects/meson/test cases/cmake/1 basic/subprojects/cmMod
|   - toolchain file:           /home/daniel/projects/meson/test cases/cmake/1 basic/build/subprojects/cmMod/__CMake_build/CMakeMesonToolchainFile.cmake
|   - preload file:             /home/daniel/projects/meson/test cases/cmake/1 basic/build/meson-private/data/preload.cmake
|   - trace args:               --trace-expand --trace-format=json-v1 --no-warn-unused-cli --trace-redirect=cmake_trace.txt
|   - disabled policy warnings: [CMP0025, CMP0047, CMP0056, CMP0060, CMP0065, CMP0066, CMP0067, CMP0082, CMP0089, CMP0102]
```

Do you have a minimal test case, or can you at least provide a log file with only the meson failure? The link you provided is not really helpful since I have to dig through 10k lines of output.

**Comment by Neumann-A** [comment]
Just download the artifacts. Those contain the failing port. Alternativley, try building toml++ with meson. I will provide the exact error msg as soon as i get on a pc.

**Comment by Neumann-A** [comment]
[config-x64-linux-dbg-out.log](https://github.com/mesonbuild/meson/files/7729524/config-x64-linux-dbg-out.log)
[meson-native-x64-linux.log](https://github.com/mesonbuild/meson/files/7729526/meson-native-x64-linux.log)
[meson-native-x64-linux-debug.log](https://github.com/mesonbuild/meson/files/7729528/meson-native-x64-linux-debug.log)

**Comment by mensinda** [comment]
Should be fixed with #9737.
